src/pocketdb/consensus/social/Blocking.hpp

```diff
@@ -86,14 +86,101 @@ namespace PocketConsensus
+    {
+    public:
+        BlockingConsensus_checkpoint_multiple_blocking(int height) : BlockingConsensus(height) {}
+        ConsensusValidateResult Validate(const CTransactionRef& tx, const BlockingRef& ptx, const PocketBlockRef& block) override
+        {
+            // Base validation with calling block or mempool check
+            if (auto[baseValidate, baseValidateCode] = SocialConsensus::Validate(tx, ptx, block); !baseValidate)
+                return {false, baseValidateCode};
+
+            // Double blocking in chain
+            if (PocketDb::ConsensusRepoInst.ExistBlocking(
+                    *ptx->GetAddress(),
+                    IsEmpty(ptx->GetAddressTo()) ? "" : *ptx->GetAddressTo(),
+                    IsEmpty(ptx->GetAddressesTo()) ? "[]" : *ptx->GetAddressesTo()
+                ))
+            {
+                if (!CheckpointRepoInst.IsSocialCheckpoint(*ptx->GetHash(), *ptx->GetType(), SocialConsensusResult_DoubleBlocking))
+                    return {false, SocialConsensusResult_DoubleBlocking};
+            }
+
+            return Success;
+        }
+        ConsensusValidateResult Check(const CTransactionRef& tx, const BlockingRef& ptx) override
+        {
+            if (auto[baseCheck, baseCheckCode] = SocialConsensus::Check(tx, ptx); !baseCheck)
+                return {false, baseCheckCode};
+
+            // Check required fields
+            if (IsEmpty(ptx->GetAddress())) return {false, SocialConsensusResult_Failed};
+            if (IsEmpty(ptx->GetAddressTo()) && IsEmpty(ptx->GetAddressesTo())) return {false, SocialConsensusResult_Failed};
+            if (!IsEmpty(ptx->GetAddressTo()) && !IsEmpty(ptx->GetAddressesTo())) return {false, SocialConsensusResult_Failed};
+
+            // Blocking self
+            auto blockingaddresses = IsEmpty(ptx->GetAddressTo()) ? "" : *ptx->GetAddressTo();
+            blockingaddresses.append(IsEmpty(ptx->GetAddressesTo()) ? "" : *ptx->GetAddressesTo());
+            if ((blockingaddresses).find(*ptx->GetAddress()) != std::string::npos)
+                return {false, SocialConsensusResult_SelfBlocking};
+
+            return Success;
+        }
+    protected:
+        ConsensusValidateResult ValidateBlock(const BlockingRef& ptx, const PocketBlockRef& block) override
+        {
+            for (auto& blockTx: *block)
+            {
+                if (!TransactionHelper::IsIn(*blockTx->GetType(), {ACTION_BLOCKING, ACTION_BLOCKING_CANCEL}))
+                    continue;
+
+                auto blockPtx = static_pointer_cast<Blocking>(blockTx);
+
+                if (*blockPtx->GetHash() == *ptx->GetHash())
+                    continue;
+
+                if (*ptx->GetAddress() == *blockPtx->GetAddress() && *ptx->GetAddressTo() == *blockPtx->GetAddressTo() &&
+                    *ptx->GetAddressesTo() == *blockPtx->GetAddressesTo())
+                    return {false, SocialConsensusResult_ManyTransactions};
```

Comment:
need check IsEmpty(GetAddressTo) and IsEmpty(GetAddressesTo) for ptx and blockPtx

Reply:
Also situation:
- In mempool exist transaction A -> B
- New transaction A -> [ B ]

Reply:
Edited